tests/gocase/util/server.go

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package util
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-redis/redis/v9"
+	"github.com/stretchr/testify/require"
+	"net"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+)
+
+type KvrocksServer struct {
+	t   *testing.T
+	cmd *exec.Cmd
+	r   *redis.Client
+
+	clean func()
+}
+
+func (s *KvrocksServer) Client() *redis.Client {
+	return s.r
+}
+
+func (s *KvrocksServer) Close() {
+	require.NoError(s.t, s.cmd.Process.Kill())
+	require.EqualError(s.t, s.cmd.Wait(), "signal: killed")
+	s.clean()
+}
+
+func StartServer(t *testing.T, configs map[string]string) (*KvrocksServer, error) {
+	b := os.Getenv("KVROCKS_BIN_PATH")
+	cmd := exec.Command(b)
+
+	addr, err := findFreePort()
+	if err != nil {
+		return nil, err
+	}
+	configs["bind"] = addr.IP.String()
+	configs["port"] = fmt.Sprintf("%d", addr.Port)
+
+	dir := os.Getenv("GO_CASE_WORKSPACE")
+	require.NoError(t, err)
+	dir, err = os.MkdirTemp(dir, "Server-*")
+	require.NoError(t, err)
+	configs["dir"] = dir
+
+	f, err := os.CreateTemp(dir, "*.conf")
```

Comment:
I think `kvrocks.conf` is enough rather than a random string? We can locate it more easy if it has a fixed name.

Reply:
Good point! Will push a commit for it.

Reply:
Updated.